### PR TITLE
[Merged by Bors] - feat: port/CategoryTheory.Limits.Cones

### PR DESCRIPTION
**Pull request by mattrobball**
---

- [x] depends on: #2332
- [x] depends on: #2326
- [x] depends on: #2299

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2332
* leanprover-community/mathlib4#2326
* leanprover-community/mathlib4#2299
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2332~~
* ~~leanprover-community/mathlib4#2326~~
* ~~leanprover-community/mathlib4#2299~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by jcommelin** [comment]
These git histories are getting a bit out of hand for reviewing purposes. Do you think you can rebase some of these PRs on current master?

File: Mathlib/CategoryTheory/Limits/Cones.lean [hunk 1]
```diff
@@ -676,155 +701,156 @@ variable {F : J ⥤ C} {G : J ⥤ C} (H : C ⥤ D)
 
 open CategoryTheory.Limits
 
+/- Porting note: dot notation on the functor is broken for `mapCone` -/
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Cones.lean:704]: Please move this note next to a manifestation of the problem.

**Review by jcommelin — APPROVED** [review]
bors d+

File: Mathlib/CategoryTheory/Limits/Cones.lean (hunk 1 above)
Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/Cones.lean]: 👍

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**